src/Microsoft.AspNet.Mvc.ModelBinding/Internal/ModelMetadataAttributeHelper.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Internal
+{
+    internal class ModelMetadataAttributeHelper
+    {
+        internal IEnumerable<Attribute> GetModelMetadataAttributesOnType(Type type, IEnumerable<Attribute> attributes)
+        {
+            var modelMedatadataType = type.GetTypeInfo().GetCustomAttribute<ModelMetadataTypeAttribute>();
+            if (modelMedatadataType != null)
+            {
+                var modelMedatadataAttributes = modelMedatadataType.MetadataBuddyType.GetTypeInfo().GetCustomAttributes();
+
+                //If same attribute exists in both attributes and modelMedatadataAttributes collection, 
+                //pick from attributes collection(view model)
+                attributes = attributes.Concat(modelMedatadataAttributes).GroupBy(a => a.GetType())
+                          .Select(grp => grp.First());
```

Comment:
Take a look at `ModelAttributes` - it may need some tweaking for your scenario, but let's keep this code together.